## Insurance data insights - 
group by function and pivot table

In [12]:
import pandas as pd

In [13]:
ins = pd.read_csv("insurance.csv")

In [14]:
ins.head(5)

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [15]:
ins.shape

(1338, 7)

In [19]:
ins.isna().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

In [16]:
ins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


###### Insight : 1

This dataset lists insurance details of total of 1338 indivisuals along with 7 different attributes. It states the insurance charges along with their respective features.

In [20]:
ins.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [26]:
ins.groupby(["smoker","sex","region"])["charges"].mean("charges")

smoker  sex     region   
no      female  northeast     9640.426984
                northwest     8786.998679
                southeast     8440.205552
                southwest     8234.091260
        male    northeast     8664.042222
                northwest     8320.689321
                southeast     7609.003587
                southwest     7778.905534
yes     female  northeast    28032.046398
                northwest    29670.824946
                southeast    33034.820716
                southwest    31687.988430
        male    northeast    30926.252583
                northwest    30713.181419
                southeast    36029.839367
                southwest    32598.862854
Name: charges, dtype: float64

In [30]:
pd.pivot_table(ins, values="charges", index="smoker", columns="sex", aggfunc='mean',)

sex           female          male
smoker                            
no       8762.297300   8087.204731
yes     30678.996276  33042.005975

In [32]:
pd.pivot_table(ins, values="charges", index=["smoker", "sex"], columns="region", aggfunc='mean',)

region            northeast     northwest     southeast     southwest
smoker sex                                                           
no     female   9640.426984   8786.998679   8440.205552   8234.091260
       male     8664.042222   8320.689321   7609.003587   7778.905534
yes    female  28032.046398  29670.824946  33034.820716  31687.988430
       male    30926.252583  30713.181419  36029.839367  32598.862854

###### Insight : 2
After reviewing primary descriptive analysis "smoker" seems to have major role for estimateing insurance charges.

### List comprehension

###### form 0 to 1000 write the numbers which are divisible by 53

In [6]:
list1 = [i for i in range(1000) if i%53 == 0]

In [7]:
print(list1, end = " ")

[0, 53, 106, 159, 212, 265, 318, 371, 424, 477, 530, 583, 636, 689, 742, 795, 848, 901, 954] 

###### Find the square of list elements

In [8]:
list2 = [1,2,3,4,5,6,7]

In [9]:
sqr = [x*x for x in list2]

In [10]:
sqr

[1, 4, 9, 16, 25, 36, 49]